codegen-lite/src/test/resources/software/amazon/awssdk/codegen/lite/regions/service-metadata-provider.java

```diff
@@ -0,0 +1,372 @@
+import software.amazon.awssdk.regions.servicemetadata.RedshiftServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.RekognitionServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.ResourceGroupsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.Route53ServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.Route53domainsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.RuntimeLexServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.RuntimeSagemakerServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.S3ServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SagemakerServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SdbServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SecretsmanagerServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.ServerlessrepoServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.ServicecatalogServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.ServicediscoveryServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.ShieldServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SmsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SnowballServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SnsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SqsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SsmServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.StatesServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.StoragegatewayServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.StreamsDynamodbServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.StsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SupportServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.SwfServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.TaggingServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.TranslateServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.WafRegionalServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.WafServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.WorkdocsServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.WorkmailServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.WorkspacesServiceMetadata;
+import software.amazon.awssdk.regions.servicemetadata.XrayServiceMetadata;
+
+@Generated("software.amazon.awssdk:codegen")
+@SdkPublicApi
+public final class ServiceMetadataProvider {
+    private static final Map<String, ServiceMetadata> SERVICE_METADATA = Collections.unmodifiableMap(Stream.of(
```

Comment:
Can we use ImmutableMap.Builder here? I feel like that would cut down on the verbosity.

Reply:
Can do.